<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/pstrawse/groovy_workspace/drc_configuration/xacro/V1_sim.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="V1" xmlns:xacro="http://www.ros.org/wiki/xacro">
    <link name="Trunk">
        <inertial>
            <mass value="41.9"/>
            <origin rpy="0 0 0" xyz="0.255 0 -0.0981"/>
            <inertia ixx="1.18918135833" ixy="0.0" ixz="0.0" iyy="1.42217329167" iyz="0.0" izz="1.35591891667"/>
        </inertial>
        <visual>
            <geometry>
                <mesh filename="package://V1/meshes/reduced_polygon_meshes/wj3.stl"/>
                <!-- <box size="${width} ${height} ${depth}"/> -->
            </geometry>
            <origin rpy="0 -1.57079632679 3.14159265359" xyz="0.03157 0.246 -0.29842"/>
            <!-- <origin rpy="${origin_rpy}" xyz="${origin_xyz}"/> -->
            <material name="White"/>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://V1/meshes/reduced_polygon_meshes/wj3.stl"/>
            </geometry>
            <origin rpy="0 -1.57079632679 3.14159265359" xyz="0.03157 0.246 -0.29842"/>
        </collision>
    </link>
    <link name="RightShoulderExtensor">
        <inertial>
            <mass value="2.65"/>
            <origin rpy="0 0 0" xyz="0 -0.01204 -0.251"/>
            <inertia ixx="0.0137181666667" ixy="0.0" ixz="0.0" iyy="0.0105028333333" iyz="0.0" izz="0.0148064333333"/>
        </inertial>
        <visual>
            <geometry>
                <mesh filename="package://V1/meshes/reduced_polygon_meshes/aj1_Right.stl"/>
                <!-- <box size="${width} ${height} ${depth}"/> -->
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <!-- <origin rpy="${origin_rpy}" xyz="${origin_xyz}"/> -->
            <material name="Gold"/>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://V1/meshes/reduced_polygon_meshes/aj1_Right.stl"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        </collision>
    </link>
    <link name="RightShoulderAdductor">
        <inertial>
            <mass value="2.87"/>
            <origin rpy="0 0 0" xyz="0.040 0.005 -0.02"/>
            <inertia ixx="0.0122024325" ixy="0.0" ixz="0.0" iyy="0.0175559325" iyz="0.0" izz="0.018176745"/>
        </inertial>
        <visual>
            <geometry>
                <mesh filename="package://V1/meshes/reduced_polygon_meshes/aj2.stl"/>
                <!-- <box size="${width} ${height} ${depth}"/> -->
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <!-- <origin rpy="${origin_rpy}" xyz="${origin_xyz}"/> -->
            <material name="White"/>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://V1/meshes/reduced_polygon_meshes/aj2.stl"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        </collision>
    </link>
    <link name="RightShoulderRotator">
        <inertial>
            <mass value="2.575"/>
            <origin rpy="0 0 0" xyz="0.02 -0.004 -0.273"/>
            <inertia ixx="0.0227863895833" ixy="0.0" ixz="0.0" iyy="0.0213984645833" iyz="0.0" izz="0.00932579166667"/>
        </inertial>
        <visual>
            <geometry>
                <mesh filename="package://V1/meshes/reduced_polygon_meshes/aj3_Right.stl"/>
                <!-- <box size="${width} ${height} ${depth}"/> -->
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <!-- <origin rpy="${origin_rpy}" xyz="${origin_xyz}"/> -->
            <material name="White"/>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://V1/meshes/reduced_polygon_meshes/aj3_Right.stl"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        </collision>
    </link>
    <link name="RightElbowExtensor">
        <inertial>
            <mass value="2.367"/>
            <origin rpy="0 0 0" xyz="0.0270 -0.0802 0"/>
            <inertia ixx="0.004941705" ixy="0.0" ixz="0.0" iyy="0.00469849291667" iyz="0.0" izz="0.00530133991667"/>
        </inertial>
        <visual>
            <geometry>
                <mesh filename="package://V1/meshes/reduced_polygon_meshes/aj4.stl"/>
                <!-- <box size="${width} ${height} ${depth}"/> -->
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <!-- <origin rpy="${origin_rpy}" xyz="${origin_xyz}"/> -->
            <material name="Gold"/>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://V1/meshes/reduced_polygon_meshes/aj4.stl"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        </collision>
    </link>
    <joint name="RightShoulderExtensor" type="revolute">
        <origin rpy="1.57079632679 -1.57079632679 3.14159265359" xyz="0 0.246 0"/>
        <axis xyz="0 0 1"/>
        <parent link="Trunk"/>
        <child link="RightShoulderExtensor"/>
        <limit effort="190" lower="-3.1415926518" upper="3.1415926518" velocity="5.89"/>
        <controller_gains Kd="400" Kp="20000"/>
        <dynamics damping="1.0" friction="0.0"/>
    </joint>
    <joint name="RightShoulderAdductor" type="revolute">
        <origin rpy="1.57079632679 0 3.14159265359" xyz="0 0 -0.246"/>
        <axis xyz="0 0 1"/>
        <parent link="RightShoulderExtensor"/>
        <child link="RightShoulderAdductor"/>
        <limit effort="190" lower="-3.05432618925" upper="-0.1745329251" velocity="5.89"/>
        <controller_gains Kd="400" Kp="20000"/>
        <dynamics damping="1.0" friction="0.0"/>
    </joint>
    <joint name="RightShoulderSupinator" type="revolute">
        <origin rpy="-1.57079632679 0 1.57079632679" xyz="0 0 0"/>
        <axis xyz="0 0 1"/>
        <parent link="RightShoulderAdductor"/>
        <child link="RightShoulderRotator"/>
        <limit effort="65" lower="-1.5707963259" upper="1.5707963259" velocity="11.5"/>
        <controller_gains Kd="400" Kp="20000"/>
        <dynamics damping="1.0" friction="0.0"/>
    </joint>
    <joint name="RightElbowExtensor" type="revolute">
        <origin rpy="1.57079632679 0 1.57079632679" xyz="0 -0.0254 -0.33"/>
        <axis xyz="0 0 1"/>
        <parent link="RightShoulderRotator"/>
        <child link="RightElbowExtensor"/>
        <limit effort="65" lower="-1.9198621761" upper="-0.08726646255" velocity="11.5"/>
        <controller_gains Kd="400" Kp="20000"/>
        <dynamics damping="1.0" friction="0.0"/>
    </joint>
    <link name="RightWristYoke">
        <inertial>
            <mass value="0.1"/>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <inertia ixx="2.25083333333e-05" ixy="0.0" ixz="0.0" iyy="2.9175e-05" iyz="0.0" izz="8.33333333333e-06"/>
        </inertial>
        <visual>
            <geometry>
                <box size="0.03 0.01 0.051"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <material name="Gray"/>
        </visual>
        <collision>
            <geometry>
                <box size="0.03 0.01 0.051"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        </collision>
    </link>
    <link name="RightPalm">
        <inertial>
            <mass value="0.712"/>
            <origin rpy="0 0 0" xyz="0.0425 0 0"/>
            <inertia ixx="0.000679010666667" ixy="0.0" ixz="0.0" iyy="0.000514360666667" iyz="0.0" izz="0.00102201666667"/>
        </inertial>
        <visual>
            <geometry>
                <mesh filename="package://V1/meshes/reduced_polygon_meshes/palm_Right.stl"/>
                <!-- <box size="${width} ${height} ${depth}"/> -->
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
            <!-- <origin rpy="${origin_rpy}" xyz="${origin_xyz}"/> -->
            <material name="Gold"/>
        </visual>
        <collision>
            <geometry>
                <mesh filename="package://V1/meshes/reduced_polygon_meshes/palm_Right.stl"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        </collision>
    </link>
    <joint name="RightWristExtensor" type="revolute">
        <origin rpy="3.14159265359 -1.57079632679 0" xyz="0.0254 -0.2870 0"/>
        <axis xyz="0 0 1"/>
        <parent link="RightElbowExtensor"/>
        <child link="RightWristYoke"/>
        <limit effort="14" lower="-0.8726646255" upper="0.6981317004" velocity="1"/>
        <controller_gains Kd="400" Kp="20000"/>
        <dynamics damping="1.0" friction="0.0"/>
    </joint>
    <joint name="RightWrist" type="revolute">
        <origin rpy="1.57079632679 0 1.57079632679" xyz="0 0 0"/>
        <axis xyz="0 0 1"/>
        <parent link="RightWristYoke"/>
        <child link="RightPalm"/>
        <limit effort="14" lower="-0.5235987753" upper="0.43633231275" velocity="1"/>
        <controller_gains Kd="400" Kp="20000"/>
        <dynamics damping="1.0" friction="0.0"/>
    </joint>
</robot>
